services/ui-src/src/components/banners/AdminBannerProvider.tsx

```diff
@@ -37,12 +38,38 @@ export const AdminBannerProvider = ({ children }: Props) => {
     setBannerDeleting,
   } = useStore();
 
+  const fetchAllBanners = async () => {
+    setBannerLoading(true);
+    try {
+      const allBanners = await getBanners();
+      // sort by latest start date (assuming the most recent one should be active)
+      allBanners.sort(
+        (a: AdminBannerData, b: AdminBannerData) => b.startDate - a.startDate
+      );
+      setAllBanners(allBanners);
+    } catch {
+      setBannerErrorMessage(bannerErrors.GET_BANNER_FAILED);
+    } finally {
+      setBannerLoading(false);
+    }
+  };
+
   const fetchAdminBanner = async () => {
     setBannerLoading(true);
     try {
-      const currentBanner = await getBanner(ADMIN_BANNER_ID);
-      const newBannerData = currentBanner?.Item || {};
-      setBannerData(newBannerData);
+      let currentBanners = await getBanners();
+      if (currentBanners.length > 1) {
+        // limit to banners active at current time
+        currentBanners = currentBanners.filter((banner: AdminBannerData) =>
+          checkDateRangeStatus(banner.startDate, banner.endDate)
+        );
+        // sort by latest start date (assuming the most recent one should be active)
+        currentBanners.sort(
+          (a: AdminBannerData, b: AdminBannerData) => b.startDate - a.startDate
+        );
+      }
+      const bannerData = currentBanners[0] || {};
+      setBannerData(bannerData);
```

Comment:
Non-blocking suggestion: This is definitely a style consideration and not a correctness one, but the `if length > 0` check isn't doing much, and I like the look of `.find(...)` over `.filter(...)[0]`. So I might have written something more like this:
```suggestion
      // Find the most recent currently-active banner
      const currentBanner = currentBanners
        .sort((a, b) => b.startDate - a.startDate)
        .find((banner) => checkDateRangeStatus(bannner.startDate, banner.endDate));
      setBannerData(currentBanner ?? {});
```

Note that if you apply this suggestion as-is, Typescript will begin to complain that `{}` isn't a real banner. I think the right move might be to remove the `?? {}` altogether, since `setBannerData()` claims to accept `undefined`, but I haven't tested that or looking into possible consequences.

(The reason that TS isn't complaining right now is because it assumes that `currentBanners[0]` will always be defined, which is not true)